lib/csv_export.rb

```diff
@@ -0,0 +1,70 @@
+module CsvExport
+  require 'csv'
+  require 'zip'
+
+  # generates a csv from the given model data
+  def generate_csv(data)
+    objects = data.first
+    columns = data.size == 1 ? objects.first.attributes.keys : data[1]
```

Comment:
Why do they need to take the same kind of input? They're two different methods doing two different things; I wouldn't necessarily expect them to take the same input. My feelings on a single input called `table` doesn't really change much, it's still two different things in an array; I'd prefer it to still be two parameters, documentation notwithstanding :smile:. We can easily keep the `map` from `generate_zip` and just use the splat operator to turn an array into two parameters to be passed into `generate_csv`, i.e. `generate_csv(*table)`.